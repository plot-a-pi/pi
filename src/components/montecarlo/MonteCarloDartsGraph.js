import React from 'react';
import ScatterGraph from '../graphs/ScatterGraph';
import { generateMonteCarloData } from '../../data/montecarlo';
const monteCarloData = generateMonteCarloData(1000);
const dartsArray = monteCarloData.dartsArray;
const sampleSizVersusPi = monteCarloData.sampleSizeVersusPiArray;

const MonteCarloDartsGraph = () => {

  return (
    <>
      <ScatterGraph data={dartsArray} xMax={1} yMax={1} xLabel={'x'} yLabel={'y'} title={'Darts'} />
      <ScatterGraph data={sampleSizVersusPi} xMax={1000} yMax={5} xLabel={'Sample Size'} yLabel={'Pi Approximation'} title={'Monte Carlo Pi Approximation'} />
    </>
  );
};

export default MonteCarloDartsGraph;
