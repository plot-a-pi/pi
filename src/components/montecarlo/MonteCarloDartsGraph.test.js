import React from 'react';
import { shallow } from 'enzyme';
import MonteCarloDartsGraph from './MonteCarloDartsGraph';

describe('MonteCarloDartsGraph component', () => {
  it('renders MonteCarloGraph', () => {
    const wrapper = shallow(<MonteCarloDartsGraph />);
    expect(wrapper).toMatchSnapshot();
  });
});
