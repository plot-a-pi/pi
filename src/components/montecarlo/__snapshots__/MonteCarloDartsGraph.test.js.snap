// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonteCarloDartsGraph component renders MonteCarloGraph 1`] = `
<Fragment>
  <h1>
    Hello MonteCarloDartsplot with inscribed circle in a square
  </h1>
</Fragment>
`;
